File: storage/data_transpiled/at_coder/arc007/C/3526777.java
public static int cSaving ( String C ) {
  int n = C . length ( ) ;
  char [ ] pattern = new char [ n ] ;
  for ( int k = 0 ;
  k < n ;
  k ++ ) {
    pattern [ k ] = ( 'x' + C + C ) . toCharArray ( ) ;
  }
  int ans = n ;
  for ( int bits = 1 ;
  bits < 1 << n ;
  bits ++ ) {
    char [ ] tmp = new char [ n ] ;
    StringBuffer string = new StringBuffer ( ) ;
    int tv_last = - 1 ;
    for ( int k = 0 ;
    k < n ;
    k ++ ) {
      if ( ( bits & ( 1 << k ) ) != 0 ) {
        tv_last = k ;
        tmp [ k ] = pattern [ k ] ;
      }
    }
    for ( int i = 0 ;
    i < tmp . length ;
    i ++ ) {
      if ( tmp [ i ] == 'o' ) {
        string . append ( 'o' ) ;
      }
      else {
        string . append ( 'x' ) ;
      }
    }
    if ( string . length ( ) > tv_last ) {
      ans = Math . min ( ans , tmp . length ) ;
    }
  }
  return ans ;
}

File: data/data_original/at_coder/arc007/C/3526777.py
def c_saving(C):
    n = len(C)
    # ??????????????????????(???n??)
    pattern = [('x' * k + C + C)[:n * 2] for k in range(n)]
    ans = n  # n???????????

    for bits in range(1, 1 << n):
        # bits?2????????2**k(k?????)???1??????
        # pattern[k]?????????(??1??????)
        tmp = []  # ??????????????
        string = ''  # ??????
        tv_last = -1
        for k in range(n):
            if bits & (1 << k) != 0:
                tv_last = k
                tmp.append(pattern[k])

        for s in zip(*tmp):
            # ???????????????????????????????
            # 'o'?????1???????????????????????
            if 'o' in s:
                string += 'o'
            else:
                string += 'x'

        # ????????????????????????'o'???
        # ?????????????????????
        # ????????????????????????????
        if string[tv_last:] == 'o' * len(string[tv_last:]):
            # print(*tmp, '|', string)
            ans = min(ans, len(tmp))
    return ans

C = input()
print(c_saving(C))